debezium-server-redis/src/main/java/io/debezium/server/redis/RedisMemoryThreshold.java

```diff
@@ -26,55 +25,79 @@ public class RedisMemoryThreshold {
     private static final String INFO_MEMORY = "memory";
     private static final String INFO_MEMORY_SECTION_MAXMEMORY = "maxmemory";
     private static final String INFO_MEMORY_SECTION_USEDMEMORY = "used_memory";
-
-    private static final Supplier<Boolean> MEMORY_OK = () -> true;
+    private static long accumulatedMemory = 0L;
+    private static long previouslyUsedMemory = 0L;
+    private static long totalProcessed = 0;
 
     private RedisClient client;
-
-    private int memoryThreshold;
-
-    private long memoryLimit;
-
-    private Supplier<Boolean> isMemoryOk;
+    private long memoryLimit = 0;
+    private long maximumMemory = 0;
 
     public RedisMemoryThreshold(RedisClient client, RedisStreamChangeConsumerConfig config) {
         this.client = client;
-        this.memoryThreshold = config.getMemoryThreshold();
         this.memoryLimit = 1024L * 1024 * config.getMemoryLimitMb();
-        if (memoryThreshold == 0 || memoryTuple(memoryLimit) == null) {
-            disable();
-        }
-        else {
-            this.isMemoryOk = () -> isMemoryOk();
-        }
     }
 
-    public boolean check() {
-        return isMemoryOk.get();
+    /**
+     * @param client
+     */
+    public void setRedisClient(RedisClient client) {
+        this.client = client;
     }
 
-    private boolean isMemoryOk() {
-        Tuple2<Long, Long> memoryTuple = memoryTuple(memoryLimit);
-        if (memoryTuple == null) {
-            disable();
+    /**
+     * Redis Enterprise samples the Redis database memory usage in an interval. Since the throughput of Redis is very big,
+     * it is impossible to rely on delayed memory usage reading to prevent OOM.
+     * In order to protect the Redis database and throttle down the sink we have created this mechanism:...
+     * @param extraMemory   - Estimated size of a single record.
+     * @param bufferSize    - Number of records in a batch.
+     * @param bufferFillRate - Rate in which memory can be filled.
+     * @return
+     */
+    public boolean checkMemory(long extraMemory, int bufferSize, int bufferFillRate) {
+        Tuple2<Long, Long> memoryTuple = memoryTuple();
+
+        maximumMemory = memoryTuple.getItem2();
+
+        if (maximumMemory == 0) {
+            totalProcessed += bufferSize;
+            LOGGER.debug("Total Processed Records: {}", totalProcessed);
```

Comment:
If `maximumMemory` is set to `0`, I assume we are not doing any memory check, and is there a case where `totalProcessed` here would just continue to grow until it overflows potentially?  If so, I wonder if it would be better to avoid incrementing and simply just return `true` only?

Reply:
This is correct is maximumMemory we are not doing any checks. The totalProcessed is long so I dobt that Debezium will ever process 9,223,372,036,854,775,807 records without restart